### PR TITLE
Docs - Add Readme

### LINKED ISSUE #35: Add readme to iota-sdk repo
The iota-sdk needs a readme file with the following contents:

- [x] Status
- [x] License
- [x] One Paragraph of the project description.
- [x] Getting Started (for each language)
    - [x] Prerequisites
    - [x] Installing
- [x] Usage
- [x] Contributing
- [x] Link to API Ref

### PR DESCRIPTION
**Pull request by lucas-tortora**
# Description of change

**Replacement PR for https://github.com/iotaledger/iota-sdk/pull/525 as something went wrong with a rebase there.**
* Added a README file to the repository
* Updated the Python and Node.js READMEs to match
 
## Links to any relevant issues

fixes https://github.com/iotaledger/guild-devx/issues/35

## Type of change

- Documentation Fix

## How the change has been tested

The readmes were viewed from the repository.

## Change checklist

- [x] I have made corresponding changes to the documentation

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,5 +1,233 @@
...
+    let wallet = Wallet::builder()
+        .with_secret_manager(SecretManager::Stronghold(secret_manager))
+        .with_client_options(client_options)
+        .with_coin_type(SHIMMER_COIN_TYPE)
+        .finish()
+        .await?;
+
+    // Generate a mnemonic and store it in the Stronghold vault.
+    // INFO: It is best practice to back up the mnemonic somewhere secure.
+    let mnemonic = wallet.generate_mnemonic()?;
```
Comment (Thoralf-M) [inline README.md:177]:
```suggestion
    let mnemonic = wallet.generate_mnemonic()?;
    println!("Creating a wallet with mnemonic:\n'{mnemonic}'");
```

File: bindings/nodejs/README.md [hunk 2]
```diff
@@ -64,43 +103,89 @@ async function run() {
...
+const walletOptions: WalletOptions = {
+    storagePath: `Alice`, // A name to associate with the created account.
+    clientOptions: {
+        nodes: ['https://api.testnet.shimmer.network'], // The node to connect to.
+    },
+    coinType: CoinType.Shimmer,
+    secretManager: {
+        // Setup Stronghold secret manager
+        stronghold: {
+            snapshotPath: 'vault.stronghold', //  The path to store the account snapshot.
```
Comment (Thoralf-M) [inline bindings/nodejs/README.md:123]:
```suggestion
            snapshotPath: 'vault.stronghold', // The path to store the account snapshot.
```

File: README.md [hunk 3]
```diff
@@ -1,5 +1,234 @@
...
+
+## Features
+
+- **Client module**: The `client` module in the IOTA SDK offers low-level functions that allow you to have
+  fine-grained control over your interactions with Shimmer nodes. It provides access to the underlying API endpoints
+  and enables advanced operations such as custom message construction and direct communication with the network.
+
+- **Wallet module**: The `wallet` module in the IOTA SDK provides high-level functions for managing accounts, generating
+  addresses, creating transactions, and interacting with the Shimmer network. It offers a user-friendly interface for
+  developers to build applications on the Shimmer network.
```
Comment (Thoralf-M) [inline README.md]: This is somewhat duplicated with the first section, maybe merge them and remove one?

File: README.md [hunk 4]
```diff
@@ -1,5 +1,234 @@
...
+
+## Getting Started
+
+### Install the IOTA SDK
+
+To start using the IOTA SDK in your Rust project, you can include the following dependencies in your `Cargo.toml` file:
+
+```toml
+[dependencies]
+iota-sdk = { branch = "develop" }
```
Comment (Thoralf-M) [inline README.md]: This doesn't work without a gh link

File: README.md [hunk 5]
```diff
@@ -1,5 +1,234 @@
...
+### Usage
+
+#### Client
+
+To use the Client module, you simply need to create a `Client`.
+
+```rust
+use iota_sdk::client::{
+    Client,
+};
```
Comment (Thoralf-M) [inline README.md:128]:
```suggestion
use iota_sdk::client::Client;
```

File: README.md [hunk 6]
```diff
@@ -1,5 +1,234 @@
...
+
+let client = Client::builder()
+    .with_node('https://api.testnet.shimmer.network')? // Insert your node URL here
+    .finish()
+    .await?;
+        
+let info = client.get_info().await?;
+println!("Node Info: {info:?}")
+
+Ok(())
```
Comment (Thoralf-M) [inline README.md]: There is no main function and also not other function at all, so one can't just run it like this and the `Ok(())` is useless (with a main function that returns a Result it's not)

File: README.md [hunk 7]
```diff
@@ -1,5 +1,234 @@
...
+    },
+    wallet::{ClientOptions, Result, Wallet},
+};
+use std::path::PathBuf;
+
+#[tokio::main]
+async fn main() -> Result<()> {
+    // Setup Stronghold secret manager.
+    let secret_manager = StrongholdSecretManager::builder()
+        .password("vault.stronghold") // A password to encrypt the stored data.WARNING: Never hardcode passwords in production code.
```
Comment (Thoralf-M) [inline README.md:162]:
```suggestion
        .password("vault.stronghold") // A password to encrypt the stored data. WARNING: Never hardcode passwords in production code.
```

File: bindings/nodejs/README.md [hunk 8]
```diff
@@ -1,46 +1,85 @@
...
+## Table of contents
+
+- [Requirements (only for building the binary yourself)](#requirements-only-for-building-the-binary-yourself)
+- [Getting Started](#getting-started)
+    - [Install the IOTA SDK](#installing-the-iota-sdk)
+    - [Usage](#usage)
+      - [Client](#client)
+      - [Wallet](#wallet)
+      - [Examples](#examples)
+- [Available Scripts](#available-scripts)
```
Comment (Thoralf-M) [inline bindings/nodejs/README.md:12]: Only the first 3 things are scripts, below are just files/folders

File: bindings/nodejs/README.md [hunk 9]
```diff
@@ -1,46 +1,85 @@
...
+Please ensure you have installed the [required dependencies for the library for Rust code](/README.md#requirements), as
+well as the following:
 
 - Python < 3.11
 - Yarn v1
 
-## Installation
+### Windows
+
+On Windows, you will also need an LLVM. Our workflow uses
```
Comment (Thoralf-M) [inline bindings/nodejs/README.md:36]:
```suggestion
On Windows, you will also need LLVM. Our workflow uses
```

File: README.md [hunk 10]
```diff
@@ -1,5 +1,234 @@
...
+The IOTA SDK is a Rust-based project that provides a convenient and efficient way to interact with nodes in the
+Shimmer and IOTA networks running
+the [Stardust protocol](https://wiki.iota.org/shimmer/develop/explanations/what-is-stardust). It consists of two main
+modules: `client` and `wallet`.
+
+The `client` module is stateless. It aims to provide more flexibility and access to low-level functions.
+
+The `wallet` module is stateful, with a standardized interface for developers to build applications involving value
+transactions. It uses high-level functions that simplify everyday operations. It can optionally interact
+with [IOTA Stronghold](https://github.com/iotaledger/stronghold.rs/) for seed handling, storage, and state backup.
```
Comment (lucas-tortora) [inline README.md:14]:
```suggestion
```

File: README.md [hunk 11]
```diff
@@ -1,5 +1,234 @@
...
+
+```rust
+use iota_sdk::client::{
+    Client,
+};
+
+#[tokio::main]
+async fn main() -> Result<()> {
+    let client = Client::builder()
+        .with_node('https://api.testnet.shimmer.network')? // Insert your node URL here
```
Comment (Thoralf-M) [inline README.md]: From where do you copy this always? :trollface: Single quote doesn't work in Rust
```suggestion
        .with_node("https://api.testnet.shimmer.network")? // Insert your node URL here
```

**Review by Thoralf-M — APPROVED** [review]

**Review by thibault-martinez — APPROVED** [review]

**Pull request merged**